### PR TITLE
Fix the player position synchronization between client and server 

### PR DESCRIPTION
**Pull request by Thomas-Harit**
…rue on initialize

### CODE REVIEW

**Review by STMiki — DISMISSED** [review]

File: cubic-server/Entity.hpp [hunk 1]
```diff
@@ -72,6 +73,8 @@ class Entity : public utility::SharedFromThis<Entity> {
     NODISCARD virtual const Vector3<double> &getPosition() const;
     NODISCARD virtual Vector2<uint8_t> &getRotation();
     NODISCARD virtual const Vector2<uint8_t> &getRotation() const;
+    NODISCARD virtual Vector2<float> getRotationDegree();
+    NODISCARD virtual const Vector2<float> getRotationDegree() const;
```
Comment (huntears) [inline cubic-server/Entity.hpp:77]: Please return a reference for those two functions

**Review by huntears — CHANGES_REQUESTED** [review]
Really good overhaul, just one small thing to fix.

File: cubic-server/Entity.hpp (hunk 1 above)
Comment (Thomas-Harit) [reply cubic-server/Entity.hpp:77]: Those functions return a locally declared variable, can't reference that :pepe_sad:

Comment (huntears) [reply cubic-server/Entity.hpp:77]: Then just create only one function like this:
```suggestion
    NODISCARD virtual Vector2<float> getRotationDegree() const;
```

**Review by STMiki — APPROVED** [review]

**Review by huntears — APPROVED** [review]

**Pull request merged**